src/test/java/org/json/junit/XMLTest.java

```diff
@@ -1247,6 +1249,39 @@ public void testIndentComplicatedJsonObjectWithArrayAndWithConfig(){
+
+        Throwable throwable = assertThrows(JSONException.class, new ThrowingRunnable() {
+
+            @Override
+            public void run() throws Throwable {
+                XML.toJSONObject(wayTooLongMalformedXML, XMLParserConfiguration.ORIGINAL.withMaxNestingDepth(42));
+            }
+        });
+
+        assertTrue("Wrong throwable thrown: not expecting message <" + throwable.getMessage() + ">", throwable.getMessage().startsWith("Maximum nesting depth of"));
+
+        final String perfectlyFineXML = "<Test>\n" +
+            "  <employee>\n" +
+            "    <name>sonoo</name>\n" +
+            "    <salary>56000</salary>\n" +
+            "    <married>true</married>\n" +
+            "  </employee>\n" +
+            "</Test>\n";
+
+        throwable = assertThrows(JSONException.class, new ThrowingRunnable() {
+
+            @Override
+            public void run() throws Throwable {
+                XML.toJSONObject(perfectlyFineXML, XMLParserConfiguration.ORIGINAL.withMaxNestingDepth(1));
+            }
+        });
+
+        assertTrue("Wrong throwable thrown: not expecting message <" + throwable.getMessage() + ">", throwable.getMessage().startsWith("Maximum nesting depth of"));
+    }
```

Comment:
An additional test would be useful, to verify that an XML having the permitted nesting depth can be converted.